goku.js

```diff
@@ -295,13 +318,47 @@
         return;
       }
 
-      if (!this.isPaused) {
-        // console.log('step', this.id, this.elapsedTime);
+      if (this.isPaused && this.pausingIndex > -1) {
+        return;
       }
 
+      var task;
+      var properties;
+      var key;
+
+      if (this.isPaused && this.pausingIndex < 0) {
+        // Set the current style to element.style
+        this.pausingIndex = this.playingIndex;
+        var styles = getComputedStyle(this.element);
+        task = this.queue[this.playingIndex];
+        properties = task.properties;
+
+        for (key in properties) {
+          this.element.style[key] = styles[key];
+        }
+
+        return;
+      }
+
+      if (!this.isPaused && this.pausingIndex > -1) {
+        this.pausingIndex = -1;
+        task = this.queue[this.playingIndex];
```

Comment:
I don't get it. It seems that we never use `pausingIndex` for anything. Why do we need this variable?